src/libmongoc/src/mongoc/mongoc-scram-private.h

```diff
@@ -123,6 +123,346 @@ _mongoc_sasl_prep_required (const char *str);
 char *
 _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err);
 
+/* returns how many bytes a UTF-8 character is. */
+int
```

Comment:
Use `size_t` to represent lengths, not `int`, for consistency with `strlen()`, `sizeof()`, etc.

Reply:
I ended up changing every instance of `char` (UTF-8 input or output code unit) into `uint8_t`, every instance of `unsigned int` (Unicode code point) into `uint32_t`, and every instance of a measure of `int` or another measure of length into `size_t`. This is true for everything besides the functions that already existed prior to the changes in this PR (e.g. `_mongoc_sasl_prep` and `_mongoc_sasl_prep_impl`), as I didn't want to change the interfaces that were already established.